src/charm.py

```diff
@@ -4,27 +4,289 @@
+        """Handle the config changed event."""
+        # Only execute on unit leader
+        if not self.unit.is_leader():
+            return
+
+        # Set the cluster name in the peer relation databag if it is not already set
+        peer_data = self._peers.data[self.app]
+
+        if not peer_data.get("cluster-name"):
+            peer_data["cluster-name"] = (
+                self.config.get("cluster-name") or f"cluster_{generate_random_hash()}"
+            )
+
+    def _on_leader_elected(self, _):
+        """Handle the leader elected event.
+
+        Set config values in the peer relation databag.
+        """
+        peer_data = self._peers.data[self.app]
+
+        required_passwords = ["root-password", "server-config-password", "cluster-admin-password"]
+
+        for required_password in required_passwords:
+            if not peer_data.get(required_password):
+                logger.debug(f"Setting {required_password}")
+                password = generate_random_password(PASSWORD_LENGTH)
+                peer_data[required_password] = password
+
+    def _on_mysql_pebble_ready(self, event):
+        """Pebble ready handler.
+
+        Define and start a pebble service and bootstrap instance.
+        """
+        if not self._is_peer_data_set:
+            self.unit.status = WaitingStatus("Waiting for leader election.")
+            logger.debug("Leader not ready yet, waiting...")
+            event.defer()
+            return
+
+        container = event.workload
+
+        if not container.exists(CONFIGURED_FILE):
+            # First run setup
+            self.unit.status = MaintenanceStatus("Initialising mysqld")
+            try:
+                # Allow mysql instances to reach each other
+                self._mysql.patch_dns_searches(self.app.name)
+
+                # Run mysqld for the first time to
+                # bootstrap the data directory and users
+                logger.debug("Initialising instance")
+                self._mysql.initialise_mysqld()
+
+                # Add the pebble layer
+                container.add_layer(MYSQLD_SERVICE, self._pebble_layer, combine=False)
+                container.restart(MYSQLD_SERVICE)
+                logger.debug("Waiting for instance to be ready")
+                self._mysql._wait_until_mysql_connection()
+
+                logger.info("Configuring instance")
+                # Configure all base users and revoke
+                # privileges from the root users
+                self._mysql.configure_mysql_users()
+                # Configure instance as a cluster node
+                self._mysql.configure_instance()
+
+            except (
+                MySQLConfigureInstanceError,
+                MySQLConfigureMySQLUsersError,
+                MySQLInitialiseMySQLDError,
+                MySQLPatchDNSSearchesError,
+            ) as e:
+                self.unit.status = BlockedStatus("Unable to configure instance")
```

Comment:
Not a very helpful error message, it would be useful to give the operator insight into what they would need to do to unbork the system.

Reply:
We have a guideline to keep status messages very high level, and when needed, users can check details with `juju debug-log`

Reply:
I thought `debug-log` will be going away soon.

Reply:
Don't now about that. But even if it was the case, wouldn't it be super seed by something better? (I don't think we are ditching logging all together)

Reply:
Debug log isn't going anywhere as such; you're on track here keeping status simple and logging out the detail.